sw/device/silicon_creator/lib/drivers/watchdog.c

```diff
@@ -5,42 +5,59 @@
 #include "sw/device/silicon_creator/lib/drivers/watchdog.h"
 
 #include "sw/device/lib/base/abs_mmio.h"
+#include "sw/device/silicon_creator/lib/base/sec_mmio.h"
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+#include "sw/device/silicon_creator/lib/drivers/otp.h"
 
 #include "aon_timer_regs.h"
 #include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "otp_ctrl_regs.h"
 #include "pwrmgr_regs.h"
 
 enum {
   kBase = TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR,
   kPwrMgrBase = TOP_EARLGREY_PWRMGR_AON_BASE_ADDR,
-  // The AON domain is always clocked at 200 Hz.
-  // TODO(lowRISC/opentitan#7385): update this after the AON frequency is
-  // formally defined in the project.
-  kAonTimerRate = 200000,
 };
 
-void watchdog_init(uint32_t timeout_ms) {
+void watchdog_init(lifecycle_state_t lc_state) {
+  // Determine the threshold cycles.
+  uint32_t threshold = 0;
+  switch (lc_state) {
+    case kLcStateTest:
+    case kLcStateRma:
+      // Set the watchdog bite threshold to a high default value when in test
+      // and RMA lifecycle states.
+      threshold = UINT32_MAX;
+      break;
+    default:
+      // In all other states read the watchdog bite threshold from OTP.
+      threshold =
+          otp_read32(OTP_CTRL_PARAM_ROM_WATCHDOG_BITE_THRESHOLD_CYCLES_OFFSET);
+  }
+
+  watchdog_enable(threshold);
+}
+
+void watchdog_enable(uint32_t threshold) {
   // Tell pwrmgr we want watchdog reset events to reset the chip.
-  abs_mmio_write32(
+  sec_mmio_write32(
```

Comment:
I wonder whether it would be better to create a driver for the pwrmgr (maybe copy the dif) and call it here rather than access the registers directly here.

Reply:
In general I think you are right. In this specific case I think a driver is probably overkill. If we start doing more with the pwrmgr then we could reconsider.